autotest/gcore/vsis3.py

```diff
@@ -2300,21 +2300,26 @@ def test_vsis3_read_credentials_config_file():
 # a non default profile
 
 
-def test_vsis3_read_credentials_config_file_non_default():
+def test_vsis3_read_credentials_config_file_non_default_profile(tmpdir, monkeypatch):
 
     if gdaltest.webserver_port == 0:
         pytest.skip()
 
     gdal.SetConfigOption('AWS_SECRET_ACCESS_KEY', '')
     gdal.SetConfigOption('AWS_ACCESS_KEY_ID', '')
-
-    gdal.SetConfigOption('CPL_AWS_CREDENTIALS_FILE', '/vsimem/aws_credentials')
-    gdal.SetConfigOption('AWS_CONFIG_FILE', '/vsimem/aws_config')
+    gdal.SetConfigOption('CPL_AWS_CREDENTIALS_FILE', None)
+    gdal.SetConfigOption('AWS_CONFIG_FILE', None)
     gdal.SetConfigOption('AWS_DEFAULT_PROFILE', 'myprofile')
 
+    if sys.platform == 'win32':
+        monkeypatch.setenv('USERPROFILE', str(tmpdir))
```

Comment:
there's a test failure on Windows. I'd suspect (but didn't check) that monkeypatch.setenv() on Windows doesn't manage to alter the native C environmennt, and just modifies the Python os.environ. Perhaps you should just use gdal.SetConfigOption(). Actually you can use ```with gdaltest.config_option(key, value):``` that will automatically save & restore the original value

Reply:
I didn't have a Windows platform to test on so indeed it was a shot in the dark...
I didn't know about `with gdaltest.config_option`, I will try it out, thanks!

Reply:
Done in 5d663ea

Reply:
The AppVeyor builds seem to be stuck in a queue, but they turned green on my fork: https://ci.appveyor.com/project/glostis/gdal/builds/30488889

Reply:
There are OSGeo/PROJ builds being processed: https://ci.appveyor.com/project/OSGeo/proj/history . So will eventually turn green here too once they and other PR are finished. Good news that it fixes your builds.

Reply:
I didn't know the build limitation was organisation-wise, and not repo-wise.
Thanks!